app/controllers/resources_controller.rb

```diff
@@ -13,7 +13,7 @@ def new
   end
 
   def show
-    redirect_to(params.merge(:action => "edit"))
+    redirect_to :action => :edit
```

Comment:
Hence the security issue... Not sure which pramas we need in the edit action. I think instead of passing whole params hash we can filter using strong_parameters?